administration/src/components/applications/ApplicationApplicantController.tsx

```diff
@@ -25,6 +26,7 @@ const ApplicationApplicantController = (props: { providedKey: string }) => {
   })
 
   if (loading) return <CircularProgress style={{ margin: 'auto' }} />
+  else if (error?.message.includes('Collection is empty')) return <InvalidLink />
```

Comment:
Let's not rely on internal error messages. We should probably define a new GraphQLError (currently we only have InvalidEmailError as an example I think) . Also relates to our error handling issue #873 .

Reply:
Hmm.. yeah I agree... It feels like it's getting messy quickly. 🤔 
Maybe, for this PR we could keep the complexity low, by simply returning False (in the graphql endpoint) if the link is invalid (where invalid includes "Link is expired"). Then, we show the user something like 
```md
# Der Link ist ungültig.

Der Link zum Zurücksetzen Ihres Passwortes ist ungültig. Mögliche Gründe:
* Der Link ist abgelaufen. Sie können [hier](/reset-password) einen neuen Link erhalten.
* Der Link wurde fehlerhaft in den Browser übertragen. Versuchen Sie, den Link manuell aus der Email in die Adresszeile Ihres Browsers zu kopieren.
* Sie haben Ihr Passwort mithilfe des Links bereits zurückgesetzt
* Sie haben einen weiteren Link angefordert. Es ist immer nur der aktuellste Link gültig.
```

Writing this up made me release that we cannot really know which of these cases we're in, so imo it makes sense to simplify our system by returning a `false` boolean instead of throwing several errors. What do you think?

Reply:
hm but this endpoint returns application data not a boolean. So we need a separate function that just returns a boolean right?

Reply:
Maybe for this case we could just check if the error codes are equal
```js
const codesEqual = error.graphQLErrors.every((value, index, array)=>value.extensions.code === array[0].extensions.code);
```

Reply:
hm but then we need a separate Query that just returns a boolean (valid/invalid) right now we getting an ApplicationView